docs/dev/zksync-v2/l1-l2-interop.md

```diff
@@ -56,13 +47,26 @@ Fourthly, there needs to be a way to bypass spam attacks for important transacti
 For each transaction, besides supplying the transaction input data, the user also provides the following:
 
 - The queue type to use. It is either a `Deque`, `HeapBuffer` or `Heap`. `Deque` is used for sequential processing of the transactions, while `HeapBuffer` and `Heap` are used for fee-prioritized processing of transactions. For the testnet, only `Deque` is available. The details on using `HeapBuffer` and `Heap` will be available later with their release.
-- The operation tree to use. It is either `Full` or `Rollup`. The details about the operation trees will be available once the zkPorter is out. For now, only the `Full` option can be used.
 - The operator fee for the transaction as part of the transaction `value` (the amount of ETH passed with the L1 transaction).
 
 ## Priority mode
 
 If the operator fails to process the needed L1 transactions, the system enters the Priority mode. In this mode, everyone can become an operator by staking tokens. The exact details of the priority mode are still under development and will be described in more detail closer to the mainnet launch.
 
-## L2->L1 communication
+## L2 -> L1 communication
+
+L2 -> L1 communication, in contrast to L1 -> L2 communication, is based only on transferring of the information, and not on the execution transaction on the other side. It is a built-in feature, which is made up of two parts: sending a message from L2 and reading a message in L1. The first is implemented as a call to the L2 system smart contract. And the second is implemented on the L1 zkSync smart contract.
+
+### Sending message
+
+Each message sent from L2 and L1 contains the sender's address and the message itself. The length of the sent message can be arbitrarily large, but the longer the message, the more expensive it is to send. The operator will "show" each message in Ethereum, so that information about all messages is always publicly available. Because of this, messages sent to L1 are guaranteed to be read, regardless of the honesty of the operator.
```

Comment:
```suggestion
Each message sent from L2 and L1 contains the sender's address and the message itself. The length of the message can be arbitrarily large, but the longer the message, the more expensive it is. The operator will "show" each message in Ethereum, so that information about all messages is always publicly available. Because of this, messages sent to L1 are guaranteed to be read, regardless of the honesty of the operator.

Reply:
> Because of this, messages sent to L1 are guaranteed to be read, regardless of the honesty of the operator.

I am not sure that I follow this sentence. Do we really need it?

Reply:
I just wanted to say, that all messages from L2 to L1 can be read without the help of a validator. In other words, if you sent a message, then you or someone else can prove that message was sent.

Reply:
Ok, the problem was not on the execution of something by the validator. I mean, with the zkSync block we sent L2 -> L1 messages that were stored in that block and the Merkle root of the messages. Operator can't send Merkle root of messages without providing data for all leaves (in our case - messages).